metadata.yaml

```diff
@@ -0,0 +1,14 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: istio-core
```

Comment:
This probably needs to be istio-core-k8s

Reply:
Actually I think the name we came up with was "istio" not "istio-core" so this should be "istio-k8s".

Reply:
it's left as "istio-core" in lucid and I think `core` is a bit more indicative that it's not the charm you'll be relating to?, anyways I'm fine with both :D

Reply:
renamed to istio-k8s, and also merged the metadata.yaml and charmcraft.yaml files. 

Will put in a request to rename the repo in future.

Reply:
oh wait, didn't even need IS I could rename it myself.  nice